views/FinalizePage/FinalizePage.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import { useForm, SubmitHandler } from 'react-hook-form';
+import * as yup from 'yup'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { addDoc, collection } from 'firebase/firestore';
+import { useRouter } from 'next/router'
+import { Button, TextField } from '@mui/material';
+import { FormWrapper, NewTextField, WrapperForm } from './FinalizePage.styles';
+import { db } from '../../firebase/config';
+import { useSession } from 'next-auth/react';
+
+
+type Inputs = {
+  name: string,
+  lastname: string,
+  age: number,
+  country: string
+};
+
+
+
+export function FinalizePage() {
+  const router = useRouter()
+  const { data: session } = useSession()
+
+  const schema = yup.object().shape({
+    name: yup.string().required("Imie jest wymagane"),
+    lastname: yup.string().required("Nazwisko jest wymagane"),
+    age: yup.string().required("Wiek jest wymagany"),
+    country: yup.string().required('Państwo jest wymagane')
+  })
+
+  const { register, handleSubmit, formState: { errors } } = useForm<Inputs>({
+    resolver: yupResolver(schema)
+  });
+
+  const onSubmit: SubmitHandler<Inputs> = async (data) => {
+    await addDoc(collection(db, "users"), {
+      name: data.name,
+      lastname: data.lastname,
+      age: data.age,
+      country: data.country,
+      email: session?.user?.email
+    });
+    router.push('/')
+  }
+
+  return (
+    <FormWrapper>
+      <WrapperForm onSubmit={handleSubmit(onSubmit)}>
+        <NewTextField
```

Comment:
If you use MUI, please don't style components through styled components. Do it through nested styling in Wrapper component or in MUI themes